subprojects/core-api/src/main/java/org/gradle/process/BaseExecSpec.java

```diff
@@ -77,6 +77,6 @@ default Property<Boolean> getIsIgnoreExitValue() {
      *
      * @return The full command line, including the executable plus its arguments
      */
-    @ToBeReplacedByLazyProperty
-    List<String> getCommandLine();
+    @ReplacesEagerProperty
+    Provider<List<String>> getCommandLine();
```

Comment:
💭 CommandLine is now a Provider<List> since it's actually a calculated from multiple values, from executable and args (and jvm args for JavaExecAction). To set full command line you have to use commandLine(...) methods and setCommandLine(...) is removed. Also you can't use commandLine = <> anymore

Reply:
I added this in the upgrade guide, because this might lead to some code misbehaving later.